drivers/sensor/mcux_tempmon/mcux_tempmon.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2023 Centralp
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_tempmon
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+
+/* OTP Controller Analog Register 1 for calibration values */
+#define OCOTP_ANA1_ROOM_COUNT_SHIFT 20
+#define OCOTP_ANA1_ROOM_COUNT_MASK  (BIT_MASK(12) << OCOTP_ANA1_ROOM_COUNT_SHIFT)
+#define OCOTP_ANA1_HOT_COUNT_SHIFT  8
+#define OCOTP_ANA1_HOT_COUNT_MASK   (BIT_MASK(12) << OCOTP_ANA1_HOT_COUNT_SHIFT)
+#define OCOTP_ANA1_HOT_TEMP_SHIFT   0
+#define OCOTP_ANA1_HOT_TEMP_MASK    (BIT_MASK(8) << OCOTP_ANA1_HOT_TEMP_SHIFT)
+
+#define TEMPMON_ROOM_TEMP 25.0f
+
+struct mcux_tempmon_data {
+	uint8_t hot_temp;
+	uint16_t hot_cnt;
+	uint16_t room_cnt;
+	uint16_t temp_cnt;
+};
+
+struct mcux_tempmon_config {
+	TEMPMON_Type *base;
+};
+
+static int mcux_tempmon_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct mcux_tempmon_data *data = dev->data;
+	const struct mcux_tempmon_config *cfg = dev->config;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_DIE_TEMP) {
+		return -ENOTSUP;
+	}
+
+	/* Start the measure */
+	cfg->base->TEMPSENSE0 |= TEMPMON_TEMPSENSE0_MEASURE_TEMP_MASK;
+
+	/* Wait until measure is ready */
+	while (!(cfg->base->TEMPSENSE0 & TEMPMON_TEMPSENSE0_FINISHED_MASK)) {
+	}
```

Comment:
How long does the measurement take? May want to add a sleep to allow other threads to do work in the meantime

Reply:
Less than one tick (10.000 ticks/s).
I used GPIO and a scope to have a better time estimation and I got `11 µs` on my i.MX RT1052 board.